### PR TITLE
add warning to wireguard EKS/AKS tutorial

### PR DESCRIPTION
**Pull request by aaaaaaaalex**
## Description

Adds 'experimental' warning and explanation for Felix's `wireguardHostEncryptionEnabled` flag.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: security/encrypt-cluster-pod-traffic.md [hunk 1]
```diff
@@ -63,21 +63,28 @@ To install WireGuard on the default Amazon Machine Image (AMI):
    sudo yum install wireguard-dkms wireguard-tools -y
    ```
 
-Additionally, you will need to enable host-to-host encryption mode for WireGuard using the following command.
```
[the remark this answers is not in the document]
Comment (aaaaaaaalex) [reply security/encrypt-cluster-pod-traffic.md:66]: @mikestephen can you confirm on this point?

**Comment by aaaaaaaalex** [comment]
@caseydavenport 
> We also need to update these pages:
> - https://docs.projectcalico.org/reference/felix/configuration
> - https://docs.projectcalico.org/reference/resources/felixconfig

it seems the former link is missing wireguard entries altogether - should I still add mine, or possibly add all the wireguard options in?

File: reference/resources/felixconfig.md [hunk 2]
```diff
@@ -101,6 +101,7 @@ spec:
 | wireguardListeningPort             | Port used by WireGuard tunnels. Felix sets up WireGuard tunnel on each node specified by this port. Available for configuration only in the global FelixConfiguration resource; setting it per host, config-file or environment variable will not work. | 1-65535 | int | 51820 |
 | wireguardMTU                       | MTU set on the WireGuard interface created by Felix. Zero value means auto-detect. See [Configuring MTU]({{ site.baseurl }}/networking/mtu). | int | int | 0 |
 | wireguardRoutingRulePriority       | WireGuard routing rule priority value set up by Felix. If you change the default value, set it to a value most appropriate to routing rules for your nodes. | 1-32765 | int | 99 |
+| wireguardHostEncryptionEnabled     | **Experimental**: Adds host-namespace workload IP's to WireGuard's list of peers. Should **not** be enabled when WireGuard is enabled on a cluster's master node, as networking deadlock can occur. | true, false | boolean | false |
```
Comment (caseydavenport) [inline reference/resources/felixconfig.md:104]: "master" -> "control plane"

File: security/encrypt-cluster-pod-traffic.md [hunk 3]
```diff
@@ -63,21 +63,28 @@ To install WireGuard on the default Amazon Machine Image (AMI):
...
+Additionally, you may optionally enable host-to-host encryption mode for WireGuard using the following command.
 
 ```bash
-kubectl -n calico-system set env daemonset/calico-node --containers="calico-node" FELIX_WIREGUARDHOSTENCRYPTIONENABLED="true"
+calicoctl patch felixconfiguration default --type='merge' -p '{"spec": {"wireguardHostEncryptionEnabled": true}}'
 ```
+
+> **Warning**: `wireguardHostEncryptionEnabled` is an experimental flag that extends WireGuard encryption to host-network IP addresses. It is currently only supported on managed clusters such an EKS and AKS, where WireGuard *cannot* be enabled on the cluster's master node. Enabling this flag while WireGuard is enabled on the master node can lead to a broken cluster, and neworking deadlock.
```
Comment (caseydavenport) [inline security/encrypt-cluster-pod-traffic.md:72]: "such as" -> "deployed in" (we don't support non-EKS and non-AKS managed clusters, right?)

Comment (caseydavenport) [reply security/encrypt-cluster-pod-traffic.md:72]: also, master -> control plane here as well

File: security/encrypt-cluster-pod-traffic.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaaaaaaalex) [reply security/encrypt-cluster-pod-traffic.md:66]: confirmed the flag is not required on EKS with either CNI :+1:

**Comment by caseydavenport** [comment]
> it seems the former link is missing wireguard entries altogether - should I still add mine, or possibly add all the wireguard options in?

Ick, it should probably have all of them. Assuming they are indeed supported on the FelixConfiguration API.

**Review by caseydavenport — APPROVED** [review]
Many thanks for doing this, especially for documenting all of the other wireguard config options that we've missed!

**Comment by aaaaaaaalex** [comment]
no prob! these docs will help me too in the future when I forget it all :laughing:

**Pull request merged**